rero_ils/modules/acq_accounts/api.py

```diff
@@ -408,65 +390,78 @@ def get_ancestors(self):
             return [parent] + parent.get_ancestors()
         return []
 
+    def _list_object_by_pid(self, record_class, query):
+        """Get record object from search query by record pid.
+
+        :param record_class: record class
+        :param query: search query
+        :return record object.
+        """
+        for hit in query.source(['pid']).scan():
+            yield record_class.get_record_by_pid(hit.pid)
+
     def get_children(self, output=None):
         """Get children accounts related to this account.
 
-        :param output: output method. 'count', 'pids' or None
+        :param output: output method. 'count' or None
         :return a generator of children accounts (or length).
         """
-        def _list_object():
-            for hit in query.source(['pid']).scan():
-                yield AcqAccount.get_record_by_pid(hit.pid)
-
         query = AcqAccountsSearch().filter('term', parent__pid=self.pid)
         if output == 'count':
             return query.count()
-        elif output == 'pids':
-            return sorted_pids(query)
-        else:
-            return _list_object()
-
-    def get_related_invoices(self, output=None):
-        """Get invoices related to this account.
-
-        :param output: output method. 'count', 'pids' or None
-        :return a generator of related invoices (or length).
-        """
-        def _list_object():
-            for hit in query.source(['pid']).scan():
-                yield AcquisitionInvoice.get_record_by_pid(hit.pid)
-
-        query = AcquisitionInvoicesSearch()\
-            .filter('term', invoice_items__acq_account__pid=self.pid)
-        if output == 'count':
-            return query.count()
-        elif output == 'pids':
-            return sorted_pids(query)
-        else:
-            return _list_object()
+        return self._list_object_by_pid(AcqAccount, query)
 
     def get_links_to_me(self, get_pids=False):
-        """Get resources linked to this object.
+        """Record links.
 
         :param get_pids: if True list of linked pids
                          if False count of linked records
         """
-        output = 'pids' if get_pids else 'count'
-        links = {
-            'acq_order_lines': self.get_related_order_lines(output=output),
-            'acq_accounts': self.get_children(output=output),
-            'acq_invoices': self.get_related_invoices(output=output)
-        }
-        links = {k: v for k, v in links.items() if v}
+        self.get_children(output=None)
+        links = {}
+        orl_query = AcqOrderLinesSearch()\
+            .filter('term', acq_account__pid=self.pid)
+        chn_query = AcqAccountsSearch()\
+            .filter('term', parent__pid=self.pid)
+        inv_query = AcquisitionInvoicesSearch()\
+            .filter('term', invoice_items__acq_account__pid=self.pid)
+        rpt_query = AcqReceiptsSearch()\
```

Comment:
In the project, we try to user as far as possible readable variable names. It should be better to use `order_lines_query`, `children_query`, ...